packages/neos-ui-editors/src/CodeMirror/index.js

```diff
@@ -0,0 +1,54 @@
+import React, {Component, PropTypes} from 'react';
+
+import Button from '@neos-project/react-ui-components/lib/Button/';
+import Icon from '@neos-project/react-ui-components/lib/Icon/';
+import Label from '@neos-project/react-ui-components/lib/Label/';
+import I18n from '@neos-project/neos-ui-i18n';
+import CodeMirrorWrap from './CodeMirrorWrap/index';
+
+export default class CodeMirror extends Component {
+    static propTypes = {
+        identifier: PropTypes.string.isRequired,
+        renderSecondaryInspector: PropTypes.func.isRequired,
+        commit: PropTypes.func.isRequired,
+        label: PropTypes.string.isRequired,
+        highlightingMode: PropTypes.string,
+        value: PropTypes.string
+    };
+
+    static defaultProps = {
+        highlightingMode: 'htmlmixed'
+    };
+
+    constructor(props) {
+        super(props);
+        this.handleOpenCodeEditor = this.handleOpenCodeEditor.bind(this);
+        this.handleChange = this.handleChange.bind(this);
+    }
+
+    render() {
+        const {label, identifier} = this.props;
+
+        return (
+            <div>
+                <Label htmlFor={identifier}>
+                    <Button onClick={this.handleOpenCodeEditor} style="brand">
+                        <Icon icon="pencil" padded="right" style="lighter" title="Edit"/>
+                        <I18n id={label}/>
+                    </Button>
+                </Label>
+            </div>
+        );
+    }
+
+    handleChange(newValue) {
+        this.props.commit(newValue);
+    }
+
+    handleOpenCodeEditor() {
+        this.props.renderSecondaryInspector('CODEMIRROR_EDIT', () =>
+            <CodeMirrorWrap onChange={this.handleChange} value={this.props.value}
```

Comment:
If multiple lines are needed for passing the props, it is considered best practice to do this with all of them, so `onChange` should already be placed on a new line here.

Also, those props need to have a 4-space indentation relative to the opening bracket, whereas the closing bracket needs to be aligned with the last prop (on a new line).